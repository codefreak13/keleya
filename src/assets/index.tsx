const KELEYA_LOGO = require('./Images/keleya-logo.png')
const INTRO_IMAGE = require('./Images/first-intro-image.png')
const AUTHENTICATION = require('./Images/authentication-background.jpg')
const COUCH_SMILE = require('./Images/couch_smile.jpg')
const DUEDATE = require('./Images/due-date-background.jpg')
const WORKOUT_GOAL = require('./Images/workout-goal-background-image.jpg')
const NOTIFICATIONS_IMAGE = require('./Images/notifications-background-image.jpg')

export { KELEYA_LOGO, INTRO_IMAGE, AUTHENTICATION, COUCH_SMILE, DUEDATE, WORKOUT_GOAL, NOTIFICATIONS_IMAGE }