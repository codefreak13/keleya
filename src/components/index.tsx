export { default as Button } from './Button'
export { default as Line } from './BottomLine'
export { default as Progress } from './Progress'
export { default as BackButton } from './BackButton'
export { default as TextInput } from './TextInput'
export { default as CheckBox } from './CheckBox'
export { default as DatePicker } from './DatePicker'