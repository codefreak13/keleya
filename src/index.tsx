import React from 'react';
import { StatusBar, SafeAreaView } from 'react-native';
import Screens from './navigation';
import { I18nextProvider } from 'react-i18next'
import i18n from './i18n'

export default function App() {
    return (
        <I18nextProvider i18n={i18n}>
            <SafeAreaView style={{ flex: 1, }}>
                <StatusBar
                    barStyle='dark-content'
                    backgroundColor="#EAEDF4"
                    translucent={true}
                />
                <Screens />
            </SafeAreaView>
        </I18nextProvider>
    );
}
