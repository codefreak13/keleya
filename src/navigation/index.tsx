import 'react-native-gesture-handler';
import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import ThemeProvider from '../utils/theme';

import {
  Intro,
  SignIn,
  SignUp,
  Name,
  Date,
  Workout,
  Success
} from '../screens'

export type MainStackParamList = {
  Intro: undefined;
  SignIn: undefined;
  SignUp: undefined;
  Name: undefined;
  Date: undefined;
  Workout: undefined;
  Success: undefined;
};

const MainStack = createStackNavigator<MainStackParamList>();

export default function App() {

  React.useEffect(() => {
    const bootstrapAsync = async () => {
      // let userToken: any;

      // try {
      //   userToken = await AsyncStorage.getItem('token');
      //   userToken && dispatch(persist());
      // } catch (e) { }
    };
    bootstrapAsync();
  }, []);

  return (
    <ThemeProvider>
      <NavigationContainer>
        <MainStack.Navigator
          initialRouteName="Intro"
          screenOptions={{
            headerShown: false,
            headerStyle: { elevation: 0 },
            cardStyle: { backgroundColor: 'white' },
          }}>
          <MainStack.Screen name="Intro" component={Intro} />
          <MainStack.Screen name="SignIn" component={SignIn} />
          <MainStack.Screen name="SignUp" component={SignUp} />
          <MainStack.Screen name="Name" component={Name} />
          <MainStack.Screen name="Date" component={Date} />
          <MainStack.Screen name="Workout" component={Workout} />
          <MainStack.Screen name="Success" component={Success} />
        </MainStack.Navigator>
      </NavigationContainer>
    </ThemeProvider>
  );
}
