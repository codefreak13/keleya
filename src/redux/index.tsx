import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import {
    authSlice,
} from './reducers';
import thunkMiddleware from 'redux-thunk';

export const store = configureStore({
    reducer: {
        auth: authSlice
    },
    middleware: [
        ...getDefaultMiddleware({ immutableCheck: false, serializableCheck: false }),
        thunkMiddleware,
    ],
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
