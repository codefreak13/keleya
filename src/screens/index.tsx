export { default as Intro } from './Intro'
export { default as SignIn } from './SignIn'
export { default as SignUp } from './SignUp'
export { default as Name } from './Name'
export { default as Date } from './Date'
export { default as Workout } from './Workout'
export { default as Success } from './Success'
