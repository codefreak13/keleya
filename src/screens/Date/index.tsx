import React, { FC, useMemo, useState } from 'react'
import { View, Image, Text, } from 'react-native'
import { NavigationProp, ParamListBase } from '@react-navigation/native';
import { useTranslation } from 'react-i18next';
import { Formik } from "formik";
import * as Yup from "yup";

import createStyles from './styles';
import { useTheme } from '../../utils/theme';
import { DUEDATE } from '../../assets'
import { Button, Line, BackButton, DatePicker } from '../../components'

interface IProps {
    navigation: NavigationProp<ParamListBase>
}

const Date: FC<IProps> = ({ navigation }) => {
    const { t } = useTranslation()
    const theme = useTheme();
    const styles = useMemo(() => createStyles(theme), [theme]);
    const { bodyStyle, buttonStyle, imageStyle, textStyle, details, error } = styles

    const validationSchema = Yup.object().shape({
        date: Yup.date()
            .required(t('date:date_validation'))
            .label('date')
    });
    return (
        <Formik
            initialValues={{ date: '' }}
            validationSchema={validationSchema}
            onSubmit={() => { navigation.navigate('Workout') }}
        >
            {({ touched, errors, handleChange, handleSubmit, values }) => (
                <View style={bodyStyle}>
                    <Image source={DUEDATE} style={imageStyle} />
                    <Text style={textStyle}>{t('date:title')}</Text>

                    <View style={details}>
                        <DatePicker
                            date={values.date}
                            action={handleChange('date')}
                        />
                        {touched.date && errors.date && (
                            <Text style={error}>{errors.date}</Text>
                        )}
                    </View>
                    <Button title={t('common:btn_title')} buttonColor={!values.date || errors.date ? 'grey' : 'dark'} buttonStyle={buttonStyle} onPress={handleSubmit} />
                    <BackButton navigation={navigation} />
                    <Line />
                </View>
            )}
        </Formik>
    )
}

export default Date
