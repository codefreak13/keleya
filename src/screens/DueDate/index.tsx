import React, { FC, useMemo, useState } from 'react'
import { View, Image, Text, } from 'react-native'
import { NavigationProp, ParamListBase } from '@react-navigation/native';
import { Formik } from "formik";
import * as Yup from "yup";

import createStyles from './styles';
import { useTheme } from '../../utils/theme';
import { DUEDATE } from '../../assets'
import { Button, Line, BackButton, DatePicker } from '../../components'

interface IProps {
    navigation: NavigationProp<ParamListBase>
}

const validationSchema = Yup.object().shape({
    date: Yup.date()
        .required('Please choose date')
        .label('date')
});

const DueDate: FC<IProps> = ({ navigation }) => {
    const theme = useTheme();
    const styles = useMemo(() => createStyles(theme), [theme]);
    const { bodyStyle, buttonStyle, imageStyle, textStyle, details, error } = styles

    const [state, setstate] = useState({
        date: '',
    });

    return (
        <View style={bodyStyle}>
            <Image source={DUEDATE} style={imageStyle} />
            <Text style={textStyle}>When is your baby due, Sam?</Text>

            <Formik
                initialValues={{ date: '' }}
                validationSchema={validationSchema}
                onSubmit={() => { navigation.navigate('WorkoutTimes') }}
            >
                {({ touched, errors, handleChange, handleBlur, handleSubmit, values }) => (
                    <>
                        <View style={details}>
                            <DatePicker
                                date={values.date}
                                action={handleChange('date')}
                            />
                            {touched.date && errors.date && (
                                <Text style={error}>{errors.date}</Text>
                            )}
                        </View>
                        <Button title='Continue' buttonColor={!values.date || errors.date ? 'grey' : 'dark'} buttonStyle={buttonStyle} onPress={handleSubmit} />
                    </>
                )}
            </Formik>
            <BackButton navigation={navigation} />
            <Line />
        </View>
    )
}

export default DueDate
