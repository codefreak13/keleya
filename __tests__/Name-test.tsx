import 'react-native'
import React from 'react';
import renderer from 'react-test-renderer';
import { Name } from '../src/screens';

jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key: any) => key })
}));

const createTestProps = (props: Object) => ({
    navigation: {
        navigate: jest.fn()
    },
    ...props
});

it('Name snapshot', () => {
    let props: any = createTestProps({});
    const snap = renderer.create(
        <Name {...props} />
    ).toJSON();
    expect(snap).toMatchSnapshot()
})