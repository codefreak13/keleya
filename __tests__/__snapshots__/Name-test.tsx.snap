// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/Images/couch_smile.jpg",
      }
    }
    style={
      Object {
        "flex": 5,
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#4a4a4a",
        "fontSize": 35,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    name:title
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "width": "75%",
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
        showsVerticalScrollIndicator={false}
      >
        <View>
          <TextInput
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="name:name_placeholder"
            placeholderTextColor="#9b9b9b"
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "borderBottomWidth": 0.7,
                  "borderColor": "#9b9b9b",
                  "color": "#4a4a4a",
                  "fontSize": 27,
                  "marginVertical": 16,
                  "padding": 14,
                  "paddingLeft": 39,
                  "width": "100%",
                },
                undefined,
              ]
            }
            textAlign="center"
            value=""
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderRadius": 10,
          "padding": 20,
          "width": "75%",
        },
        Object {
          "marginBottom": 39,
        },
        Object {
          "backgroundColor": "#9b9b9b",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 35,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      common:btn_title
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "position": "absolute",
        "top": 59,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï„˜
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 10,
        "bottom": 10,
        "height": 8,
        "position": "relative",
        "width": "35%",
      }
    }
  />
</View>
`;
